nion/swift/ExportDialog.py

```diff
@@ -239,45 +239,133 @@ def export_clicked(display_items: typing.Sequence[DisplayItem.DisplayItem], view
     def cancel(self) -> bool:
         return True
 
-
 class ExportSVGHandler:
+    width_value_line_edit: Declarative.UIWidget
+    height_value_line_edit: Declarative.UIWidget
     def __init__(self, display_size: Geometry.IntSize) -> None:
-        self.width_model = Model.PropertyModel(display_size.width)
-        self.height_model = Model.PropertyModel(display_size.height)
-
-        self.int_converter = Converter.IntegerToStringConverter()
+        self.display_size = display_size
+        self.aspect_ratio = display_size.width / display_size.height
+        self.width_model = Model.PropertyModel(float(display_size.width))
+        self.height_model = Model.PropertyModel(float(display_size.height))
+        self.unit_model = Model.PropertyModel(2)  # Default to pixels (index 2)
+        self.int_converter = Converter.FloatToStringConverter()
+        self.previous_unit_index = 2  # Default to pixels (index 2)
 
         u = Declarative.DeclarativeUI()
-        width_row = u.create_row(u.create_label(text=_("Width (in)"), width=80), u.create_line_edit(text="@binding(width_model.value, converter=int_converter)"), spacing=12)
-        height_row = u.create_row(u.create_label(text=_("Height (in)"), width=80), u.create_line_edit(text="@binding(height_model.value, converter=int_converter)"), spacing=12)
-        main_page = u.create_column(width_row, height_row, spacing=12, margin=12)
-
+        self.width_line_edit = u.create_line_edit(
+            text="@binding(width_model.value, converter=int_converter)",
+            on_text_edited="value_updated",
+            background_color="white"
+            ,name="width_value_line_edit"
+        )
+        self.height_line_edit = u.create_line_edit(
+            text="@binding(height_model.value, converter=int_converter)",
+            on_text_edited="value_updated",
+            background_color="white",
+            name="height_value_line_edit"
+        )
+        width_row = u.create_row(
+            u.create_label(text=_("Width:"), width=80),
+            self.width_line_edit,
+            spacing=12
+        )
+        height_row = u.create_row(
+            u.create_label(text=_("Height:"), width=80),
+            self.height_line_edit,
+            spacing=12
+        )
+        # Combo box for units
+        self.units_combo_box = u.create_combo_box(
+            items=[_("Inches"), _("Centimeters"), _("Pixels")],
+            current_index="@binding(unit_model.value)",
+            on_current_index_changed="unit_changed"
+        )
+        units_row = u.create_row(
+            u.create_label(text=_("Units:"), width=80),
+            self.units_combo_box,
+            spacing=12
+        )
+        main_page = u.create_column(width_row, height_row, units_row, spacing=12, margin=12)
         self.ui_view = main_page
 
+    def value_updated(self, widget: Declarative.UIWidget, text: str) -> None:
```

Comment:
Hmmm. I seem to have missed this in a bunch of other code, but let's start now!

Widgets passed to declarative callback functions are the actual widgets defined in `UserInterface.py`. So this function should be:

`def value_updated(self, widget: UserInterface.LineEditWidget, text: str) -> None:`

This goes for all similar cases in this code.